src/main/java/core/basesyntax/controller/LoginController.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.controller;
+
+import core.basesyntax.exception.AuthenticationException;
+import core.basesyntax.lib.Injector;
+import core.basesyntax.model.Driver;
+import core.basesyntax.security.AuthenticationService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class LoginController extends HttpServlet {
+    private static final String DRIVER_ID = "id";
+    private static final Injector injector = Injector.getInstance("core.basesyntax");
+    private final AuthenticationService authenticationService
+            = (AuthenticationService) injector.getInstance(AuthenticationService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("WEB-INF/views/login.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        try {
+            Driver driver = authenticationService.login(login, password);
+            HttpSession session = req.getSession();
+            session.setAttribute(DRIVER_ID, driver.getId());
+        } catch (AuthenticationException e) {
+            req.setAttribute("errorMsg", e.getMessage());
+            req.getRequestDispatcher("WEB-INF/views/login.jsp").forward(req, resp);
+            return;
```

Comment:
```suggestion
```

Reply:
After some testing of errors I found out that you are wrong.
https://stackoverflow.com/questions/2123514/java-lang-illegalstateexception-cannot-forward-sendredirect-create-session

Reply:
Partially wrong to be precise. Redirect after this return will be failed in filter but it is redundant operation since before return we can forward user straight to login page without unnecessary checking user in filter.